src/FluentCheck.ts

```diff
@@ -231,3 +252,24 @@ class FluentCheckAssert<Rec extends ParentRec, ParentRec extends {}> extends Flu
       new FluentResult(false)
   }
 }
+
+class FluentCheckGenerator<Rec extends ParentRec, ParentRec extends {}> extends FluentCheck<Rec, ParentRec> {
+  constructor(
+    readonly parent: FluentCheck<ParentRec, any>,
+    config: FluentConfig,
+    readonly generator: (seed: number) => () => number,
+    readonly seed?: number
+  ) {
+    super(parent, config)
+    if (seed !== undefined)
+      this.prng.seed = seed
+    else
+      this.prng.seed = Math.floor(Math.random() * 0x100000000)
+    this.prng.generator = generator(this.prng.seed)
```

Comment:
This should be instantiated only when the scenario is run so that two runs of the same seeded scenario give the same result.

Reply:
I've provided the same feedback in my review. I think this can be shown with a test, although we are surely missing a way to trace all the generated `testCases`.